project1/templates/search.html

```diff
@@ -0,0 +1,138 @@
+<!DOCTYPE html>
+<html>
+<head>
+
+<meta name="viewport" content="width=device-width, initial-scale=1">
+
+  <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='search_style.css') }}">
+
+
+
+ <link href="//maxcdn.bootstrapcdn.com/bootstrap/4.1.1/css/bootstrap.min.css" rel="stylesheet" id="bootstrap-css">
+  <script src="//maxcdn.bootstrapcdn.com/bootstrap/4.1.1/js/bootstrap.min.js"></script>
+  <script src="//cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+
+
+<script>
+  function apisearch() {
+
+    var request = new XMLHttpRequest();
+    var res = document.forms['res']['search'].value;
+    request.open("POST","/api/search/");
+    request.setRequestHeader("Content-Type", "application/json");
+    //when sending data to webserver, it must be a sting: using stringify
+    request.send(JSON.stringify({
+      "search": res
+    }));
+
+    //onload function processes request that we receive.
+    request.onload = () => {
+      if (request.status === 200) {
+        var data = JSON.parse(request.responseText)
+        let str = ""
+        //List to store each individual book detail.
+        listt = []
+
+
+        for (var i = 0; i < data["author"].length; i++) {
+
+          var a = `<img src="http://covers.openlibrary.org/b/isbn/${data["isbn"][i]}-M.jpg" width = "100" height = "150" >`
+
+          str = str + "<tr>" + "<td>" + a + "</td>"+
+          "<td>" + "<a href = javascript:bookPage(" + i + ")>" + data["isbn"][i] + "</a>" + "</td>" + 
+          "<td>" + data["title"][i] + "</td>" +
+          "<td>" + data["author"][i] + "</td>" +
+          "<td>" + data["year"][i] + "</td>" + "</tr>"
+
+          listt.push(data["isbn"][i])
+
+        }
+
+        document.getElementById("tab").innerHTML = str;
+
+
+      } else {
+        console.log(`error ${request.status} ${request.statusText}`)
```

Comment:
When an error has occurred, send the error info to the user as well.

Reply:
sent error info to user.